core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -297,6 +297,9 @@ object LogConfig {
   def validate(props: Properties) {
     validateNames(props)
     configDef.parse(props)
+    val config = LogConfig(props)
+    require(config.segmentSize >= config.maxMessageSize,
+      s"${LogConfig.SegmentBytesProp}=${config.segmentSize} must be larger or equal to ${LogConfig.MaxMessageBytesProp}=${config.maxMessageSize}")
```

Comment:
Doesn't this approach not work for the same reason as https://issues.apache.org/jira/browse/KAFKA-4092?